src/module.ts

```diff
@@ -1913,6 +1913,23 @@ export class Module {
     return ret;
   }
 
+  setLocalNames(funcRef: FunctionRef, localNames: string[] | null): void {
+    if (localNames == null) return;
+    let localNameMap = new Set<string>();
+    for (let i = 0, k = localNames.length; i < k; i++) {
+      let localName = localNames[i];
```

Comment:
I still wondering why not just fuse two loops into one?
```ts
  setLocalNames(funcRef: FunctionRef, func: Function): void {
    let localNameMap = new Set<string>();
    let localsByIndex = func.localsByIndex;
    for (let i = 0, k = localNames.length; i < k; i++) {
      let localName = localsByIndex[i].name;
```

Reply:
I suggest move to separate function `binaryen._BinaryenFunctionSetLocalName` in `Module` class:
```ts
setLocalName(ref: FunctionRef, localIndex: i32, localName: string): void {
   binaryen._BinaryenFunctionSetLocalName(funcRef, localIndex, this.allocStringCached(localName));
}
```
and move to `compiler.ts` instead module:
```
// compiler.ts
  setLocalNames(funcRef: FunctionRef, func: Function): void {
    let module = this.module;
    let localNameMap = new Set<string>();
    let localsByIndex = func.localsByIndex;
    for (let i = 0, k = localNames.length; i < k; i++) {
      let localName = localsByIndex[i].name;
      if (localNameMap.has(localName)) {
        let repeat = 0;
        while (localNameMap.has(`${localName}_${repeat}`)) {
          repeat++;
        }
        localName = `${localName}_${repeat}`;
      }
      localNameMap.add(localName);
      module.setLocalName(funcRef, i, localName);
    }
 }
```

Reply:
It sounds like great

Reply:
I simplify the `Module.ts` and move all logic into `Function#finalize` because it is similar as add debug info into source map.

Reply:
nice!